### PR TITLE
Feature/story by slug

### PR DESCRIPTION
**Pull request by digitalbase**

### CODE REVIEW

**Review by pablopunk — APPROVED** [review]

**Comment by pablopunk** [comment]
@digitalbase I think we could quickly setup a vercel project for this repo with an access token so we can test PRs

File: modules/Stories/Stories.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ type Props = {
 const Stories: FunctionComponent<Props> = ({ stories }) => (
     <div>
         {stories.map((story) => (
-            <Link key={story.id} href={`/story/${story.id}`} passHref>
+            <Link key={story.id} href={story.slug} passHref>
```
Comment (oleg-semyonov) [inline modules/Stories/Stories.tsx:12]: I guess it should be `/${story.slug}` with a slash?

File: modules/Story/Story.tsx [hunk 2]
```diff
@@ -6,12 +6,30 @@ type Props = {
...
-const Story: FunctionComponent<Props> = ({ story }) => (
-    <article>
-        <h2>{story.title}</h2>
-        <h3>{story.subtitle}</h3>
-        <SlateRenderer nodes={JSON.parse(story.content)} />
-    </article>
-);
+const Story: FunctionComponent<Props> = ({ story }) => {
+    if (!story) {
+        return <></>;
```
Comment (oleg-semyonov) [inline modules/Story/Story.tsx:11]: Shouldn't it return `null` if don't want to render anything?

File: modules/Story/Story.tsx [hunk 3]
```diff
@@ -6,12 +6,30 @@ type Props = {
...
+            <h3>{subtitle}</h3>
+            <SlateRenderer nodes={JSON.parse(content)} />
+
+            {format_version === 1 && (
+                // eslint-disable-next-line react/no-danger
+                <div dangerouslySetInnerHTML={{ __html: htmlContent }} />
+            )}
+            {format_version === 3 && (
+                <SlateRenderer nodes={JSON.parse(content)} />
+            )}
```
Comment (oleg-semyonov) [inline modules/Story/Story.tsx:30]: `<SlateRenderer>` is called two times here, is this intentional?

File: modules/Stories/Stories.tsx (hunk 1 above)
Comment (pablopunk) [reply modules/Stories/Stories.tsx:12]: `story.slug` works too, it will redirect properly

File: modules/Story/Story.tsx (hunk 3 above)
Comment (oleg-semyonov) [reply modules/Story/Story.tsx:30]: Also, `JSON.parse` can be quite heavy with big objects, maybe move it to a `useMemo`?

**Pull request merged**